pkg/apis/core/v1/validation/validation.go

```diff
@@ -144,6 +158,89 @@ func ValidatePodLogOptions(opts *v1.PodLogOptions) field.ErrorList {
 	return allErrs
 }
 
+// ValidateNodeLogViewerOptions checks if options that are set are at the correct
+// value. Any incorrect value will be returned to the ErrorList.
+func ValidateNodeLogViewerOptions(opts *v1.NodeLogViewerOptions) field.ErrorList {
+	files, allErrs := validateQuery(opts.Query)
+	if files && (opts.SinceTime != nil || opts.UntilTime != nil || opts.Boot != nil || opts.TailLines != nil ||
+		len(opts.Format) > 0 || len(opts.Pattern) > 0) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), opts.Query,
+			"file query cannot be combined with options"))
+	}
+	if opts.SinceTime != nil && opts.UntilTime != nil && (opts.SinceTime.After(opts.UntilTime.Time)) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("sinceTime", "untilTime"), "",
+			"`sinceTime` cannot be after `untilTime"))
+	}
+	supportedFormats := []string{"short-precise", "json", "short", "short-unix", "short-iso", "short-iso-precise",
+		"cat", ""}
+	if err := stringInSlice(opts.Format, supportedFormats); err != nil {
+		allErrs = append(allErrs, field.NotSupported(field.NewPath("format"), opts.Format, supportedFormats))
+	}
+	if opts.Boot != nil {
+		if err := utilvalidation.IsInRange((int)(*opts.Boot), -100, 0); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("boot"), *opts.Boot, err[0]))
+		}
+	}
+	if opts.TailLines != nil {
+		if err := utilvalidation.IsInRange((int)(*opts.TailLines), 0, 100000); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("tailLines"), *opts.TailLines, err[0]))
+		}
+	}
+	if err := safeString(opts.Pattern); err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("pattern"), opts.Pattern, err.Error()))
+	}
+	return allErrs
+}
+
+func validateQuery(query []string) (bool, field.ErrorList) {
+	var files, services, totalLength int
+	allErrs := field.ErrorList{}
+
+	for _, q := range query {
+		totalLength += len(q)
+		if strings.ContainsAny(q, "/\\") {
+			files++
+		} else {
+			services++
+			if err := safeString(q); err != nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("query"), q, err.Error()))
+			}
+		}
+	}
+	if totalLength > maxTotalServiceNamesLength {
+		allErrs = append(allErrs, field.TooLongMaxLength(field.NewPath("query"), query, maxTotalServiceNamesLength))
+	}
+
+	switch {
+	case files == 0 && services == 0:
+		allErrs = append(allErrs, field.Required(field.NewPath("query"), "cannot be empty"))
+	case files > 0 && services > 0:
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), query, "cannot specify a file and service"))
+	case files > 1:
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), query, "cannot specify more than one file"))
+	}
+	return files > 0, allErrs
+}
+
+func safeString(s string) error {
+	if len(s) > maxParameterLength {
+		return fmt.Errorf("input is too long, max length is %d", maxParameterLength)
```

Comment:
Error conventions want the message to be like "must be less than x".  Don't say "input" - the field name will be listed already